cmd/gapit/common.go

```diff
@@ -111,8 +113,39 @@ func getDevice(ctx context.Context, client client.Client, capture *path.Capture,
 		return nil, log.Err(ctx, err, "Failed query list of devices for replay")
 	}
 
-	if len(paths) > 0 {
-		return paths[0], nil
+	filteredByFlags := make([]*path.Device, 0, len(paths))
+	for i := 0; i < len(paths); i++ {
+		p := paths[i]
+		o, err := client.Get(ctx, p.Path())
+		if err != nil {
+			return nil, log.Err(ctx, err, "Couldn't resolve device")
+		}
+		d := o.(*device.Instance)
```

Comment:
Should we log an Info here about the command line argument value?

Reply:
Would it not make more sense to do the device matching inside the vulkan's logic for `GetDevicesForReplay` ? That way this would work for GAPIS too.

Reply:
That part will be done in #1501. This part is like, if there are more devices to replay, the user may use `-gapir-device` to specify which one to use.